moniker_src/data_moniker_service.cpp

```diff
@@ -16,51 +22,60 @@ using grpc::Status;
 using std::string;
-using ni::data_monikers::Moniker;
+using ni::data_monikers::DataMoniker;
 using ni::data_monikers::MonikerList;
 using ni::data_monikers::MonikerWriteRequest;
+using ni::data_monikers::MonikerReadResponse;
+using ni::data_monikers::StreamWriteResponse;
 using ni::data_monikers::SidebandWriteRequest;
 using ni::data_monikers::SidebandReadResponse;
-using ni::data_monikers::MonikerReadResult;
-using ni::data_monikers::StreamWriteResponse;
 using ni::data_monikers::BeginMonikerSidebandStreamRequest;
 using ni::data_monikers::BeginMonikerSidebandStreamResponse;
+using google::protobuf::Arena;
+
+static void SysFsWrite(const std::string& fileName, const std::string& value)
+{
+    std::ofstream fout;
+    fout.open(fileName);
+    fout << value;
+    fout.close();
+}
 
-namespace ni
+namespace ni::data_monikers
 {
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    MonikerServiceImpl* MonikerServiceImpl::s_Server;
+    DataMonikerService* DataMonikerService::s_Server;
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    MonikerServiceImpl::MonikerServiceImpl()
+    DataMonikerService::DataMonikerService()
     {
-        s_Server = this;		
+        s_Server = this;
     }
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    void MonikerServiceImpl::RegisterMonikerEndpoint(string endpointName, MonikerEndpointPtr endpoint)
+    void DataMonikerService::RegisterMonikerEndpoint(string endpointName, MonikerEndpointPtr endpoint)
     {
-        s_Server->_endpoints.emplace(endpointName, endpoint);	
+        s_Server->_endpoints.emplace(endpointName, endpoint);
     }
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    void MonikerServiceImpl::RegisterMonikerInstance(string endpointName, void* instanceData, Moniker& moniker)
+    void DataMonikerService::RegisterMonikerInstance(string endpointName, void* instanceData, Moniker& moniker)
     {
-        moniker.set_data_instance(reinterpret_cast<int64_t>(instanceData));	
+        moniker.set_data_instance(reinterpret_cast<int64_t>(instanceData));
         moniker.set_service_location("local");
         moniker.set_data_source(endpointName);
     }
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    void MonikerServiceImpl::InitiateMonikerList(const MonikerList& monikers, EndpointList& readers, EndpointList& writers)
+    void DataMonikerService::InitiateMonikerList(const MonikerList& monikers, EndpointList* readers, EndpointList* writers)
```

Comment:
This should probably be called InitializeMonikerList or InitMonikerList.

Reply:
Updated to **InitializeMonikerList**